corehq/apps/reports/management/commands/update_custom_recipient_type.py

```diff
@@ -0,0 +1,60 @@
+from django.core.management import BaseCommand
+from corehq.messaging.scheduling.scheduling_partitioned.models import (
+    CaseTimedScheduleInstance,
+    TimedScheduleInstance,
+    CaseAlertScheduleInstance,
+    AlertScheduleInstance,
+)
+from corehq.sql_db.util import get_db_aliases_for_partitioned_query
+from corehq.apps.data_interfaces.models import CreateScheduleInstanceActionDefinition
+from django.db.models import Q
+
+
+class Command(BaseCommand):
+    help = 'Updates a custom recipient value'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--current-type',
+            help='The current custom type which is to be updated.',
+        )
+        parser.add_argument(
+            '--new-type',
+            help='The new custom type which the current type is to be updated to.',
+        )
+
+    def handle(self, **options):
+        current_type = options.get('current_type')
+        new_type = options.get('new_type')
+
+        if not current_type or not new_type:
+            raise Exception('current-type or new-type value missing')
+
+        for db in get_db_aliases_for_partitioned_query():
+            CaseTimedScheduleInstance.objects.using(db).filter(recipient_id=current_type).update(
```

Comment:
These tables are quite large, but I'm assuming that the filtered set of rows will be small and will update quickly. Do you know if that's true? We don't want these tables to get locked for long.

Reply:
Yeah, those filtered queries are relatively small (I checked on prod; about 300 total).